src/azul/service/manifest_service.py

```diff
@@ -366,14 +385,7 @@ def for_format(cls,
         :return: a ManifestGenerator instance. Note that the protocol used for
                  consuming the generator output is defined in subclasses.
         """
-        if format_ == 'compact':
```

Comment:
I'd be open to considering the dictionary lookup approach, if you can show me (via evidence, reason or an authoritative reference) that it isn't considerable slower. It does look more concise and yes, lookup is O(1), but my rule of thumb has always been that the constant factor dominates until n=10. 1 * 100 is still more than 9 * 10. For example, is there any evidence that a `dict` literal in a function body doesn't create a new `dict` instance in every call? Is the bytecode compiler smart enough to realize that this `dict` is never modified and that it can be hoisted into a constant? We could of course extract the literal into an class attribute but then this approach loses some of its attraction.

Reply:
The bytecode compiler is not that smart and does indeed build the dictionary anew on each function invocation. This is about half as fast as if/elif/else. However if the dictionary is built once ahead of time and the function only needs to perform the lookup, it is actually about 18% faster than if/elif/else.

Source: `timeit.timeit` and `dis.dis`.

Reply:
Extraction into a class attribute is complicated because the subclasses haven't been defined defined when the class body of `ManifestService` is being processed. I'll sick with if/elif/else for now.